Makefile.am

```diff
@@ -275,7 +287,14 @@ cc_proxy_extra_dist =			\
 CHECK_DEPS += check-proxy
 
 check-proxy:
-	go test -v -race -timeout 2s $(srcdir)/proxy
+	[ -n "$$GOPATH" ] && \
+		( \
+			go test -v -race -timeout 2s $(srcdir)/proxy ;\
```

Comment:
Before landing this PR we might want to wait until #613 has landed as it looks like that one will conflict with your `check-proxy` change.